flink-yarn/src/test/java/org/apache/flink/yarn/YarnFileStageTestS3ITCase.java

```diff
@@ -173,36 +172,27 @@ private void testRecursiveUploadForYarn(String scheme, String pathSuffix) throws
         }
     }
 
-    @Test
-    @RetryOnFailure(times = 3)
-    public void testRecursiveUploadForYarnS3n() throws Exception {
+    @TestTemplate
+    @RetryOnException(times = 3, exception = Exception.class)
+    public void testRecursiveUploadForYarnS3n(@TempDir File tempFolder) throws Exception {
```

Comment:
```suggestion
    void testRecursiveUploadForYarnS3n(@TempDir File tempFolder) throws Exception {
```
also applies to other tests